net-dns/dnscrypt-proxy/files/dnscrypt-proxy.conf

```diff
@@ -0,0 +1,242 @@
+######################################################
+#                                                    #
+#    Sample configuration file for dnscrypt-proxy    #
+#                                                    #
+######################################################
+
+
+############## Resolver settings ##############
+
+## [CHANGE THIS] Short name of the resolver to use
+## Usually the only thing you need to change in this configuration file.
+## This corresponds to the first column in the dnscrypt-resolvers.csv file.
+## Alternatively, "random" (without quotes) picks a random random resolver
+## accessible over IPv4, that doesn't log and supports DNSSEC.
+
+ResolverName random
+
+
+## Full path to the list of available DNSCrypt resolvers (dnscrypt-resolvers.csv)
+## An up-to-date list is available here:
+## https://download.dnscrypt.org/dnscrypt-proxy/dnscrypt-resolvers.csv
+## and the dnscrypt-update-resolvers.sh script can be used in order to
+## automatically download and verify updates.
+
+ResolversList /usr/share/dnscrypt-proxy/dnscrypt-resolvers.csv
+
+
+## Manual settings, only for a custom resolver not present in the CSV file
+
+#ProviderName    2.dnscrypt.some.provider.name.tld
+#ProviderKey     0000:1111:2222:3333:4444:5555:6666:7777:8888:9999:AAAA:BBBB:CCCC:DDDD:EEEE:FFFF
+#ResolverAddress 111.222.333.444:56789
+
+
+
+############## Process options ##############
+
+## [NOT AVAILABLE ON WINDOWS] Run the proxy as a background process.
+## Unless you are using systemd, you probably want to change this to "yes"
+## after having verified that the rest of the configuration works as expected.
+
+Daemonize yes
+
+
+## Write the PID number to a file
+
+PidFile /var/run/dnscrypt-proxy.pid
```

Comment:
I suggest leaving the PidFile out of the configuration file, and instead passing it on the command-line (which would override the one in the config file anyway). The reason is, in the init script, you define

```
pidfile="/var/run/${SVCNAME}.pid"
```

But what happens if the user changed the `PidFile` line in the config file? The init script stops working =(
Don't ask why, but people actually do that. Instead, I would pass the existing `$pidfile` variable to the daemon:

```
command_args="${DNSCRYPT_OPTS} --pidfile="${pidfile}"
```

That way the PID file that start-stop-daemon uses is guaranteed to be the one that dnscrypt-proxy uses.

An unrelated note: the `/var/run` directory is being migrated to simply `/run`, so now is as good a time as any to make that change.